tests/asgi/middleware/test_middleware.py

```diff
@@ -142,6 +141,22 @@ async def test_should_use_segmentid_in_http_header(self, client, recorder):
         xray_header = server_response.headers[http.XRAY_HEADER]
         assert expected_root in xray_header
 
+    async def test_should_with_missing_http_host_header(self, client, recorder):
```

Comment:
```suggestion
    async def test_should_fallback_to_localhost_when_missing_http_host_header(
        self, client, recorder
    ):
```

Reply:
We use test function names that provide a brief description of the expected behaviour, as well as the scenario.